bindings/node/lib/credentialsProvider.js

```diff
@@ -1,18 +1,27 @@
 'use strict';
 
-let awsCredentialProviders = null;
-
 /**
- * Load cloud provider credentials for the user provided KMS providers.
- * Credentials will only attempt to get loaded if they do not exist
- * and no existing credentials will get overwritten.
+ * @ignore
+ * Auto credential fetching should only occur when the provider is defined on the kmsProviders map
+ * and the settings are an empty object.
  *
- * @param {Object} kmsProviders - The user provided KMS providers.
- * @returns {Promise} The new kms providers.
+ * This is distinct from a nullish provider key.
  *
- * @ignore
+ * @param {string} provider
+ * @param {object} kmsProviders
  */
-async function loadCredentials(kmsProviders) {
+function isEmptyCredentials(provider, kmsProviders) {
+  return (
+    provider in kmsProviders &&
+    kmsProviders[provider] != null &&
+    typeof kmsProviders[provider] === 'object' &&
+    Object.keys(kmsProviders[provider]).length === 0
+  );
+}
+
+let awsCredentialProviders = null;
```

Comment:
Btw, it is so nice that this package does dependency injection for the driver package – It might even be worth adapting that to this dependency as well … :)

Reply:
How come you think this is a beneficial pattern?  The DI for the driver is something I've wanted to remove for a while

Reply:
Well, from a very practical point, it’s just a necessity in cases where the driver is bundled into something but this package isn’t (which would be common in bundling scenarios given that this package is a native addon package).

But also just philosophically (and I’m pretty sure you already know this), I think DI is a great design pattern that solves more issues than it creates :slightly_smiling_face:

Reply:
Just to clarify, this thought is not directly related to the changes in this PR and should absolutely not block it :)

Reply:
We should consider this though, using it for testing too could be 🔥 

For GCP we don't plan to use an SDK (and likely not for azure either) so this would only be for AWS but still, valid use case even for one dep.